### PR TITLE
Add post check component deployment is ready

### PR DESCRIPTION
**Pull request by ajaypratap003**
## Description

Add post check component deployment is ready
Resolve https://issues.redhat.com/browse/RHOAIENG-349
## How Has This Been Tested?

- Install ODH
- Deploy basic DSC resource
- Check DSC resoruce status

## Merge criteria:

- [X] The commits are squashed in a cohesive manner and have meaningful messages.
- [ ] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [ ] The developer has manually tested the changes and verified that the changes work

### CODE REVIEW

File: components/codeflare/codeflare.go [hunk 1]
```diff
@@ -104,13 +104,18 @@ func (c *CodeFlare) ReconcileComponent(ctx context.Context, cli client.Client, l
...
+
 	// CloudServiceMonitoring handling
 	if platform == cluster.ManagedRhods {
 		if enabled {
-			// first check if the service is up, so prometheus won't fire alerts when it is just startup
-			if err := cluster.WaitForDeploymentAvailable(ctx, cli, ComponentName, dscispec.ApplicationsNamespace, 20, 2); err != nil {
-				return fmt.Errorf("deployment for %s is not ready to server: %w", ComponentName, err)
-			}
 			l.Info("deployment is done, updating monitoring rules")
 		}
```
Comment (AjayJagan) [inline components/codeflare/codeflare.go:120]: Since we move this block outside, do we need this if again here? Same for all the other files as well

**Comment by zdtsw** [comment]
@ajaypratap003 should we close this one?

**Comment by ajaypratap003** [comment]
> @ajaypratap003 should we close this one? or still valid for review

This is valid for review. Changes have done from my end. Please review and let me know if any suggestion.

**Comment by zdtsw** [comment]
i am fine with this PR the only part is, since this is now added to all offerings than just managed, we will need to fine tune the frequency and timeout.  i.e if the timeout set in QE's regression is in a lower value, test could have failed.

**Comment by ajaypratap003** [comment]
Closing this PR as Wen already picked  these changes in her PR https://github.com/opendatahub-io/opendatahub-operator/pull/1205

**Pull request closed without merging**